src/base/search/searchpluginmanager.cpp

```diff
@@ -458,30 +458,30 @@ void SearchPluginManager::update()
         if (!engineElem.isNull()) {
             QString pluginName = engineElem.tagName();
 
-            PluginInfo *plugin = new PluginInfo;
-            plugin->name = pluginName;
-            plugin->version = getPluginVersion(pluginPath(pluginName));
-            plugin->fullName = engineElem.elementsByTagName("name").at(0).toElement().text();
-            plugin->url = engineElem.elementsByTagName("url").at(0).toElement().text();
+            PluginInfo plugin;
+            plugin.name = pluginName;
+            plugin.version = getPluginVersion(pluginPath(pluginName));
+            plugin.fullName = engineElem.elementsByTagName("name").at(0).toElement().text();
+            plugin.url = engineElem.elementsByTagName("url").at(0).toElement().text();
 
             foreach (QString cat, engineElem.elementsByTagName("categories").at(0).toElement().text().split(" ")) {
                 cat = cat.trimmed();
                 if (!cat.isEmpty())
-                    plugin->supportedCategories << cat;
+                    plugin.supportedCategories << cat;
             }
 
             QStringList disabledEngines = Preferences::instance()->getSearchEngDisabled();
-            plugin->enabled = !disabledEngines.contains(pluginName);
+            plugin.enabled = !disabledEngines.contains(pluginName);
 
-            updateIconPath(plugin);
+            updateIconPath(&plugin);
 
             if (!m_plugins.contains(pluginName)) {
-                m_plugins[pluginName] = plugin;
+                m_plugins[pluginName] = new PluginInfo(plugin);
```

Comment:
It was made to prevent PluginInfo copying. Since your changes break this approach, maybe just store PluginInfo here as is (no pointer)?
Otherwise you can turn it into shared pointer above to fix memory leak without causing data copying.

Reply:
>just store PluginInfo here as is (no pointer)?

You mean refactor `QHash<QString, PluginInfo*> m_plugins` to `QHash<QString, PluginInfo> m_plugins;`? The work is a bit too much for me.

>Otherwise you can turn it into shared pointer above to fix memory leak without causing data copying.

Which shared ptr class to use?

Reply:
seems `std::unique_ptr` with `release()` is possible here

Reply:
changed.

Reply:
Really I mean to use Shared pointer (either std::shared_ptr or QSharedPointer) everywhere raw pointer to PluginInfo was used before since it's more safe.
But if you don't want to do that, your current changes will come off as a hotfix.

Reply:
I haven't look closely but why is pointer type required? can't you just use `QHash<QString, PluginInfo> m_plugins`?
Anyway I don't wish to delve deep into this, I wish to get the current approach merged.

Reply:
>I haven't look closely but why is pointer type required?

See the first comment on this line.

>Anyway I don't wish to delve deep into this, I wish to get the current approach merged.

Ok.

P.S. IMO, QScopedPointer is better choice for the approach you use here.

Reply:
>See the first comment on this line.

Still not too clear for me... I guess it's because Qt containers doesn't support move semantics... then it would be better to use STL containers here instead (just discussing here).